modules/seqexec/server/src/main/scala/seqexec/server/ghost/GHOSTController.scala

```diff
@@ -22,10 +22,85 @@ import squants.time.Time
 final case class GHOSTController[F[_]: Sync](ghostClient: GHOSTClient[F],
                                              gdsClient: GDSClient) {
   import GHOSTController._
-  val log: Logger = getLogger
+  val Log: Logger = getLogger
+
+  private sealed abstract class BundleConfig(val configName: String)
+  private final case object StandardBundleConfig extends BundleConfig("lo")
+  private final case object HighResBundleConfig extends BundleConfig("hi")
+
+  private sealed abstract class IFUNum(val ifuNum: Int) {
+    val ifuStr = s"ghost:cc:cu:ifu$ifuNum"
+  }
+  private final case object IFU1 extends IFUNum(1)
+  private final case object IFU2 extends IFUNum(ifuNum = 2)
+
+  private sealed abstract class IFUTargetType(val targetType: Int)
+  private object IFUTargetType {
+    def determineType(name: Option[String]): IFUTargetType = name match {
+      case None        => IFUTargetNone
+      case Some("Sky") => IFUTargetSky
+      case _           => IFUTargetObject
+    }
+  }
+  private final case object IFUTargetNone extends IFUTargetType(0)
+  private final case object IFUTargetSky  extends IFUTargetType(1)
+  private final case object IFUTargetObject extends IFUTargetType(2)
+
+  // TODO: Are GIAPI parameters maintained between applies? Here is a use case that requires more information:
+  // 1. We observe in standard resolution mode using both IFUs on targets.
+  // 2. We observe in high resolution mode using only IFU1 on a target.
+  // Question: Do we need to, for (2), touch IFU2 at all (i.e. to standard mode, no target)?
+  // Do we need to always set target type for IFUs? How about bundle configuration?
+  //
+  // Right now, this only generate a config if the name, RA, and dec are defined: otherwise, we get the empty config.
+  private def ifuConfig(ifuNum: IFUNum,
+                        nameOpt: Option[String],
+                        raOpt: Option[HourAngle],
+                        decOpt: Option[Angle],
+                        bundleConfig: BundleConfig): Configuration =
+      (for {
+        name <- nameOpt
+        ra   <- raOpt
+        dec  <- decOpt
+      } yield {
+        Configuration.single(ifuNum.ifuStr + ".target", IFUTargetType.determineType(nameOpt).targetType) |+|
+          Configuration.single(ifuNum.ifuStr + ".ra", ra.toDoubleDegrees) |+|
+          Configuration.single(ifuNum.ifuStr + ".dec", dec.toDoubleDegrees) |+|
+          Configuration.single(ifuNum.ifuStr + ".bundle", bundleConfig.configName)
```

Comment:
I would factor out `Configuration.single(ifuNum.ifuStr` into a local method. Stylistically I would assign the `Option[Configuration]` to a variable and then return `config.getOrElse(Configuratin.Zero)` to avoid the awkward parens around the `for`-comp.